torch/_functorch/aot_autograd.py

```diff
@@ -725,17 +737,42 @@ def inner(*flat_args):
     return inner
 
 
-# This creates a functionalized joint forwards-backwards function given both
-# the primals (to run forwards) and tangents (to run backwards).
-#
+def unpack_synthetic_bases(
+    primals: List[Any],
+    synthetic_base_info: Optional[List[Union[int, Tuple[int, torch.Tensor]]]],
+) -> List[Any]:
+    # This is only not None if our graph mutates a graph input that aliases another graph input.
+    if synthetic_base_info is None:
+        return primals
+
+    f_args_inner = []
+    for outer_idx_or_tuple in synthetic_base_info:
+        if isinstance(outer_idx_or_tuple, int):
+            f_args_inner.append(primals[outer_idx_or_tuple])
+        else:
+            outer_base_idx, view_tensor = outer_idx_or_tuple
+            outer_base = primals[outer_base_idx]
+            view_arg = gen_alias_from_base(
+                outer_base, view_tensor, view_tensor.requires_grad
+            )
+            f_args_inner.append(view_arg)
+    return f_args_inner
+
+# This creates the final function that we want to trace using make_fx(),
+# in both aot_dispatch_autograd and aot_dispatch_base.
+# The function returns has signature "traced_fn(primals: List[Any], maybe_tangents: Optional[List[Any]])",
+# where:
+# (1) if trace_joint is True, it will trace the joint fwd/bwd (by calling autograd.grad on the result of the forward)
+# (2) if trace_joint is False, it will only trace the forward.
 # It uses the metadata that was created earlier to figure out what all of the outputs to the autograd.Function.forward are:
 # (1) Which inputs received data mutations (and need to be passed as outputs into autograd.grad())
 # (2) Which outputs are aliases of inputs (and should *not* be passed as outputs into autograd.grad())
-def create_joint_forward_backward_functionalized(
+def create_forward_or_joint_functionalized(
```

Comment:
Making a function that previously did one thing do two things toggled by a bool is generally not the way to get clear, understandable code 😛

Reply:
I have audited the `trace_joint` sites and this seems acceptable

Reply:
ha - I think that despite the bool, the duplication is pretty minimal. But I could just re-write this as two separate functions calling into the shared logic